### PR TITLE
Create a JSON summary and add a compare command

### LINKED ISSUE #42: Add support for creating a `json` output and additional tooling
When running tests either repeatedly or the same tests on different system it is often useful to compare results. Currently, we store the test results in a `csv` file which is a ok, but a bit tedious to post-process in particular when trying to compare different results.

The proposal is to stash all the tests results as well as the system information and test meta data (such as start/end times etc) in a single `json` file with a reasonably well defined schema.

Once we have that we should be able to add simple tooling to read/compare multiple `json` files and generate, for example, a summary `csv` file from it.

### PR DESCRIPTION
**Pull request by rn**
This PR adds two major features:
- Record test results and system details in `SUMMARY.json` resolves #42
- A new `compare` subcommand which can show the results of multiple test runs side by side (and optionally generate a CSV from it.

In addition:
- refactors the some of the test result handling to avoid code duplication for pretty printing
- Fix the CPU reported on macOS
- Change the file extension for CSV files to lower case
- Add an optional CSV output for `rtf info`

### CODE REVIEW

**Comment by ijc** [comment]
Is `SUMMARY.json` a new ad-hoc thing we've invented or a standard(ish) preexisting thing?

It would be really handy to be compatible with something like junit, then Jenkins and such can (supposedly, I've not tried or seen it done) provide more useful charts and stuff.

**Comment by rn** [comment]
It's an adhoc thing I invented. I'll add an issue to investigate/add `junit` support. People have mentioned it in the past, but I always forget what it was called.

File: cmd/run.go [hunk 1]
```diff
@@ -176,41 +181,45 @@ func run(cmd *cobra.Command, args []string) error {
 	systemInfo := sysinfo.GetSystemInfo()
 	runConfig.SystemInfo = systemInfo
 
+	summary.ID = id
+	summary.SystemInfo = systemInfo
+	summary.Labels = labelList
+	summary.StartTime = startTime
+
```
Comment (ijc) [inline cmd/run.go]: nit: I'd be more idiomatic to declare summary as:
```
summary := local.Summary{
    ID: id,
    SystemInfo: systemInfo,
...
}
```

**Comment by rn** [comment]
Thanks, good suggestion with the idiomatic bit. Force pushed an update

**Pull request merged**